src/ImageZoom.js

```diff
@@ -115,18 +118,24 @@ export default class ImageZoom extends Component {
   }
 
   render() {
-    const { image, isMaxDimension } = this.state
+    const { image, shouldHandleZoom } = this.props
+    const { isMaxDimension, source } = this.state
+
+    const allowZooming = !isMaxDimension && shouldHandleZoom()
 
     /**
      * Take whatever attributes you want to pass the image
-     * and then override with the properties we need.
+     * and then override with the properties we need,
+     * including using state as source of truth for hi/low-res 
+     * version img src.
      * Also, disable any clicking if the component is
      * already at its maximum dimensions.
      */
     const attrs = Object.assign(
       {},
-      !isMaxDimension && { tabIndex: focusableTabIndex },
+      allowZooming && { tabIndex: focusableTabIndex },
       image,
+      { src: source },
       { style: this._getImageStyle() },
```

Comment:
With the `source` => `src` change, we can combine this with the line above it into:
```js
{ src, style: this._getImageStyle() },
```